src/lib.rs

```diff
@@ -1469,4 +1474,15 @@ mod tests {
     test_ivc_base_with::<G1, G2>();
     test_ivc_base_with::<bn256::Point, grumpkin::Point>();
   }
+
+  // SAVE_GENERATED_KEYS_TO_JSON=true cargo +nightly test test_ivc_nontrivial_with_compression_pasta --release -- --nocapture
+  #[test]
+  fn test_ivc_nontrivial_with_compression_pasta() {
```

Comment:
I would change the name to something like "test_solidity_compatibility_e2e_pasta"

Reply:
will do